webextension/background.js

```diff
@@ -1,3 +1,174 @@
+const assignManager = {
+  closeableWindows: new Set([
+    "about:startpage",
+    "about:newtab",
+    "about:home",
+    "about:blank"
+  ]),
+  menuAssignId: "open-in-this-container",
+  menuRemoveId: "remove-open-in-this-container",
+  storageArea: browser.storage.local,
+
+  init() {
+    browser.tabs.onActivated.addListener((info) => {
+      browser.tabs.get(info.tabId).then((tab) => {
+        this.calculateContextMenu(tab);
+      }).catch((e) => {
+        throw e;
+      });
+    });
+
+    browser.windows.onFocusChanged.addListener((id) => {
+      browser.tabs.query({active: true}).then((tabs) => {
+        if (tabs && tabs[0]) {
+          this.calculateContextMenu(tabs[0]);
+        }
+      }).catch((e) => {
+        throw e;
+      });
+    });
+
+    browser.runtime.onMessage.addListener((request) => {
+      if (request.neverAsk === true) {
+        this.storageArea.set({neverAsk: true});
```

Comment:
Ah, so this sets a global "never ask" preference. I thought we should start with a "Never ask again **for this site**".